### PR TITLE
Improve URL parsing

### PR DESCRIPTION
**Pull request by eliykat**
## Objective

See more info here: https://app.asana.com/0/1169444489336079/1200474469788269/f.

## Code changes

* strip dollar signs from URI before calling `getDomain()`.

`browser/jslib` will need to be bumped after this is merged.

### CODE REVIEW

**Comment by Hinton** [comment]
What happens if you use another character like `!`? Tldjs has a `isValid` function could potentially be used (return null on invalid). https://github.com/oncletom/tld.js/blob/b758887c0d01a12b6c0f048a544377e8d44368db/lib/is-valid.js#L40

File: common/src/misc/utils.ts [hunk 1]
```diff
@@ -203,7 +203,7 @@ export class Utils {
             return null;
         }
 
-        uriString = uriString.trim();
+        uriString = uriString.replace(/\$/g, '').trim();
```
Comment (cscharf) [inline common/src/misc/utils.ts:206]: I agree with Oscar, please remove this line

File: common/src/misc/utils.ts [hunk 2]
```diff
@@ -203,7 +203,7 @@ export class Utils {
...
         if (uriString === '') {
             return null;
         }
```
Comment (cscharf) [inline common/src/misc/utils.ts:209]: Down below line 213 after the `data:` check. Add another  check `tldjs.isValid()` and if not valid, then return `null`. Follow the patterns for calling `tldjs` as also seen below.

**Review by cscharf — CHANGES_REQUESTED** [review]

**Comment by eliykat** [comment]
I've updated this to use `tldjs.isValid`. Note that the type definitions for `isValid` indicate that it's deprecated in favour of `isValidHostname`. However, `isValidHostname` isn't actually implemented in the installed npm package in `node_modules`. There seems to be [some confusion about this](https://github.com/oncletom/tld.js/issues/133). I think it's a mismatch between the type definitions and the current npm package that we can ignore.

Also, `isValid` only accepts a hostname (not full url), however we already have `Util.getHostname` which doesn't use `tldjs` and which appears to parse symbols correctly, so I thought that was ok to use for this purpose.

File: common/src/misc/utils.ts [hunk 3]
```diff
@@ -221,6 +221,11 @@ export class Utils {
         if (httpUrl) {
             try {
                 const url = Utils.getUrlObject(uriString);
+                const validHostname = tldjs?.isValid != null ? tldjs.isValid(url.hostname) : true;
```
Comment (Hinton) [inline common/src/misc/utils.ts:224]: Do we need to update our noop implementation as well? (Always return true perhaps)

**Review by cscharf — APPROVED** [review]

**Pull request merged**